sambacc/ctdb.py

```diff
@@ -181,6 +182,28 @@ def read_ctdb_nodes(path: str = CTDB_NODES) -> list[str]:
     return entries
 
 
+def _svc_match(service_name: str, line: str) -> bool:
+    if not line.strip() or line.startswith("#"):
+        return False
+    first = line.split()[0]
+    return first == service_name
+
+
+def ensure_ctdb_port_in_services(port: int, path: str) -> None:
+    try:
+        with open(path, "r") as fh:
+            lines = [line.strip() for line in fh]
+    except FileNotFoundError:
+        lines = []
+    cleaned = [line for line in lines if not _svc_match("ctdb", line)]
+    cleaned.append(f"ctdb  {port}/tcp   # custom ctdb port")
+    cleaned.append(f"ctdb  {port}/udp   # custom ctdb port")
```

Comment:
Do we need an entry for udp here? According to https://wiki.samba.org/index.php/Advanced_CTDB_configuration#CTDB_port, it only opens tcp ports.

Reply:
The default file has both entries. On my system (fedora 41):
```
[jmulliga@nacho ~]$ grep ctdb /etc/services 
ctdb            4379/tcp                # CTDB
ctdb            4379/udp                # CTDB
```
I wanted it to match.